Sources/TuistKit/Cache/CacheStorageProvider.swift

```diff
@@ -3,18 +3,22 @@ import TuistCache
 import TuistCloud
 import TuistCore
 import TuistGraph
+import TuistLoader
 
 final class CacheStorageProvider: CacheStorageProviding {
-    let config: Config
+    private let config: Config
+    private let cacheDirectoryProviderFactory: CacheDirectoriesProviderFactoring
 
     init(config: Config) {
         self.config = config
+        cacheDirectoryProviderFactory = CacheDirectoriesProviderFactory()
```

Comment:
After looking at the factory pattern, I thought you'd use it for doing dependency injection and change the instance of the directories provider. However, you are not doing any dependency injection. Was that the goal? If we can easily test the classes without that, do we really need the factory?